app/src/main/java/com/example/travelapp/fragments/Graph.java

```diff
@@ -0,0 +1,79 @@
+package com.example.travelapp.fragments;
+
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+public class Graph {
+
+    HashMap<Integer, Integer>[] arr;
```

Comment:
for every function you need to check that `arr` is not null since you do not define `arr` if `n <= 0`